lib/object.c

```diff
@@ -196,21 +196,36 @@ static tool_rc tpm2_util_object_fetch_parent_from_tpk(const char *objectstr,
     tool_rc rc = tool_rc_general_error;
     BIO *input_bio = 0;
     TSSPRIVKEY_OBJ *tpk = 0;
+    BIGNUM *bn_parent = 0;
     tpm2_util_object_fetch_tssprivkey_from_file(objectstr, &input_bio, &tpk);
     if (!input_bio || !tpk) {
         goto ret;
     }
 
-    uint64_t parent;
-    int ret = ASN1_INTEGER_get_uint64(&parent, tpk->parent);
-    if (ret == 0 || parent > UINT32_MAX) {
+    bn_parent = ASN1_INTEGER_to_BN(tpk->parent, NULL);
+    if (!bn_parent) {
+        goto ret;
+    }
+
+    TPM2_HANDLE parent = 0;
+    bool is_bn_parent_negative = BN_is_negative(bn_parent);
+    if (is_bn_parent_negative) {
+        parent = ASN1_INTEGER_get(tpk->parent);
+    } else {
+        parent = BN_get_word(bn_parent);
+    }
+
+    if (parent == 0) {
         LOG_ERR("Unable to convert parent to integer/value too large");
         goto ret;
     }
     *val = parent;
     rc = tool_rc_success;
 
 ret:
+    if (bn_parent) {
```

Comment:
you don't need this check
"BN_free() frees the components of the BIGNUM, and if it was created by BN_new(), also the structure itself. BN_clear_free() additionally overwrites the data before the memory is returned to the system. If a is NULL, nothing is done."